Zend/zend_hash.h

```diff
@@ -1175,12 +1175,12 @@ static zend_always_inline void *zend_hash_get_current_data_ptr_ex(HashTable *ht,
 		HashTable *__fill_ht = (ht); \
 		Bucket *__fill_bkt = __fill_ht->arData + __fill_ht->nNumUsed; \
 		uint32_t __fill_idx = __fill_ht->nNumUsed; \
+		uint32_t __fill_old_used = __fill_idx; \
```

Comment:
You don't need this new variable. You have frozen  ``__fill_ht->nNumUsed`` so you may do:
```c
__fill_ht->nNumOfElements += __fill_idx - __fill_ht->nNumUsed;
__fill_ht->nNumUsed = __fill_idx;
```